src/litdata/streaming/dataloader.py

```diff
@@ -665,14 +665,16 @@ def state_dict(self) -> Dict[str, Any]:
                 "latest_worker_idx": self._latest_worker_idx,
             }
 
-        num_samples_yieled = [0 for _ in range(len(list(self._num_samples_yielded_combined.values())[0]))]
+        # Initialize a list to track the number of samples yielded for each dataset
+        num_samples_yieled = [0 for _ in range(len(self.dataset._datasets))]
+
         for worker_idx in self._num_samples_yielded_combined:
             for dataset_idx, samples_yieled in enumerate(self._num_samples_yielded_combined[worker_idx]):
                 num_samples_yieled[dataset_idx] += samples_yieled
 
         return {
             "dataset": self.dataset.state_dict(self.num_workers, self.batch_size, num_samples_yieled),
-            "current_epoch": self.current_epoch if self.restore else self.current_epoch - 1,
+            "current_epoch": self.current_epoch,
```

Comment:
Trying to understand why `current_epoch` is subtracted by 1. No clue so far.

Any thoughts @tchaton ?

Reply:
I made epoch start from 0 and you from 1.

Reply:
It seems that although `current_epoch` is initially set to 0, it gets updated to 1 as soon as the dataloader starts, so I think it's starting from 1.

<img src="https://github.com/user-attachments/assets/b2395ea0-c6f7-4129-b789-84c489738379" width="400" />

Reply:
Cool, let's keep yours then.